magento/magento2
# Added a validation for the dimension and enable KGS to be a possible …

The UPS integration currently only works for the unit of measurement `Inches`. European shops who want to use UPS are not able to setup their Magento integration with Kgs. 

### Description (*)
The change enables the UPS module to work interchangeably with Inch or with Cm. The change is made when the UPS mode works with API integration. 

### Related Pull Requests


### Fixed Issues (if relevant)
1. Fixes magento/magento2#38618

### Manual testing scenarios (*)
The scope of the PR changes one mode of the UPS integration and that is the Rest mode with the UPS integration:
Enable the UPS shipping method
Setup the type with the Rest mode
The shipment origin setting requires a Postcode

**Use case 1:** this currently does not return rates and that is what the fix changes
Set the Shipping origin to be European 
Set the weight unit in Kgs
Ensure the currency for the country of origin is allowed in the system
validate UPS returns some shipping methods 

**Use case 2:**: backward compliance test
Set the Shipping origin to be United States 
Set the weight unit in Lbs
Ensure the currency for the country of origin is allowed in the system
validate UPS returns some shipping methods 

**Use case 3:**: backward compliance test / the unit Kg is not authorised for US origin
Set the Shipping origin to be United States 
Set the weight unit in Kgs
Ensure the currency for the country of origin is allowed in the system
validate UPS does not return any shipping methods 

### Questions or comments
I have attempted to add integration tests. There are many in place and I was hopeful to get at least one added for the use case of unit measure 'Kg' and origin 'EU' but I found that the blocker is that these tests mocks when the API call is triggered and that means we can't add tests for this issue as far as I see it

### Contribution checklist (*)
